File: Backend/quiz/src/main/java/com/quiz/service/CategoryService.java
package com.quiz.service;

import java.util.Set;

import com.quiz.model.Category;

public interface CategoryService {
	
    //	Add Category
	public Category addCategory(Category category);
	
    //	Update Category
	public Category updateCategory(Category category);
	
	//  Get All Categories
	public Set<Category> getCategory();
	
	//  Get Single Category
	public Category getSingleCategory(Long categoryId);
	
	//  Delete Category
	public void deleteCategory(Long categoryId);

}

File: Frontend/src/services/CategoryService.js
import axios from 'axios';
import {base_url} from './base_url';
import {JwtToken} from './JwtToken';

class CategoryService{

    create(category){
        return axios.post(`${base_url}/quiz-service/category/`,category,{
            headers:{
                jwtToken: JwtToken
            }
        });
    }

    getAllCategory(){
        return axios.get(`${base_url}/quiz-service/category/`,
        {
            headers:{
                jwtToken:JwtToken
            }
        });
    }

    get(cid){
        return axios.get(`${base_url}/quiz-service/category/${cid}`,
        {
            headers:{
                jwtToken:JwtToken
            }
        });
    }

    delete(cid){
        return axios.delete(`${base_url}/quiz-service/category/${cid}`,
        {
            headers:{
                jwtToken:JwtToken
            }
        });
    }

    update(category){
        return axios.put(`${base_url}/quiz-service/category/`,category,
        {
            headers:{
                jwtToken:JwtToken
            }
        });
    }

}

export default new CategoryService();